File: 264 Count on Cantor/solution.c
#include <stdio.h>
#include <math.h>

int main() {
    int number;
    while(scanf("%d", &number) == 1) {
        int i = ceil((-1 + sqrt(1 + (number << 3))) / 2);
        int a = i + 1 - number + ((i - 1) * i >> 1);
        int b = number - ((i - 1) * i >> 1);
        if(i & 1)
            printf("TERM %d IS %d/%d\n", number, a, b);
        else
            printf("TERM %d IS %d/%d\n", number, b, a);
    }
    return 0;
}

File: 264 Count on Cantor/solution.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
        StringBuilder output = new StringBuilder();
        while(st.nextToken() == StreamTokenizer.TT_NUMBER) {
            int number = (int)st.nval;
            int i = (int)Math.ceil((-1 + Math.sqrt(1 + (number << 3))) / 2);
            int a = i + 1 - number + ((i - 1) * i >> 1);
            int b = number - ((i - 1) * i >> 1);
            output.append("TERM ").append(number).append(" IS ").append((i & 1) == 1 ? a : b).append('/').append((i & 1) == 1 ? b : a).append('\n');
		}
        System.out.print(output);
	}
}
